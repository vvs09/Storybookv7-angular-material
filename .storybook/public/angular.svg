<svg width="188" height="200" viewBox="0 0 188 200" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M94 123.4H69.3L57.6 152.6H35.9H35.8L94 22.1V22.2V0L0.899994 33.2L15.1 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M77 105.4H94V64.5L77 105.4Z" fill="#808080"/>
<g opacity="0.9">
<path d="M94 123.4H118.7L130.4 152.6H152.1H152.2L94 22.1V22.2V0L187.1 33.2L172.9 156.3L94 200V123.4Z" fill="#808080"/>
<path d="M111 105.4H94V64.5L111 105.4Z" fill="#808080"/>
</g>
</svg>
