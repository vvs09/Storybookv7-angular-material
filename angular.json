{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "storybook": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "sass"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:browser",
          "options": {
            "outputPath": "dist/storybook",
            "index": "src/index.html",
            "main": "src/main.ts",
            "polyfills": [
              "zone.js"
            ],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kb",
                  "maximumError": "1mb"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kb",
                  "maximumError": "4kb"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "buildOptimizer": false,
              "optimization": false,
              "vendorChunk": true,
              "extractLicenses": false,
              "sourceMap": true,
              "namedChunks": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "storybook:build:production"
            },
            "development": {
              "buildTarget": "storybook:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n",
          "options": {
            "buildTarget": "storybook:build"
          }
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "polyfills": [
              "zone.js",
              "zone.js/testing"
            ],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "sass",
            "assets": [
              "src/favicon.ico",
              "src/assets"
            ],
            "styles": [
              "@angular/material/prebuilt-themes/indigo-pink.css",
              "src/styles.sass"
            ],
            "scripts": []
          }
        },
        "storybook": {
          "builder": "@storybook/angular:start-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "port": 6006,
            "enableProdMode":true
          }
        },
        "build-storybook": {
          "builder": "@storybook/angular:build-storybook",
          "options": {
            "configDir": ".storybook",
            "browserTarget": "storybook:build",
            "compodoc": false,
            "outputDir": "storybook-static"
          }
        }
      }
    }
  }
}