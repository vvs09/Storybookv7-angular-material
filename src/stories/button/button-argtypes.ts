export const buttonArgTypes= {
    color: {
      control: 'radio',
      options: ['accent', 'primary', 'warn'],
      description: 'Theme color palette for the component.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'null' },
        type: {
          summary: 'string',
        },
      },
    },
    disabled: {
      control: 'boolean',
      description:
        'Whether the checkbox is disabled. This fully overrides the implementation provided by mixinDisabled, but the mixin is still required because mixinTabIndex requires it.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
    disableRipple: {
      control: 'boolean',
      description: 'Whether ripples are disabled.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
    buttonType: {
      control: 'select',
      options: [
        'mat-button',
        'mat-raised-button',
        'mat-stroked-button',
        'mat-flat-button',
      ],
      description: 'Type of buton.',
      table: {
        defaultValue: { summary: 'mat-button' },
        type: {
          summary: 'string',
        },
      },
    },
    label: {
      control: 'text',
      description: 'Button label.',
      table: {
        defaultValue: { summary: 'button' },
        type: {
          summary: 'string',
        },
      },
    },
    ariaDisabled: {
      control: 'boolean',
      description: 'aria-disabled value of the button.',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'undefined' },
        type: {
          summary: 'boolean | undefined',
        },
      },
    },
    disabledInteractive: {
      control: 'boolean',
      description:
        'Natively disabled buttons prevent focus and any pointer events from reaching the button. In some scenarios this might not be desirable, because it can prevent users from finding out why the button is disabled (e.g. via tooltip). Enabling this input will change the button so that it is styled to be disabled and will be marked as aria-disabled, but it will allow the button to receive events and focus. Note that by enabling this, you need to set the tabindex yourself if the button isnt meant to be tabbable and you have to prevent the button action (e.g. form submissions).',
      table: {
        category: 'INPUTS',
        defaultValue: { summary: 'false' },
        type: {
          summary: 'boolean',
        },
      },
    },
  }