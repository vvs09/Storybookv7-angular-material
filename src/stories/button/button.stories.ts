import type { Meta, StoryObj } from '@storybook/angular';
import { argsToTemplate, moduleMetadata } from '@storybook/angular';
import { MatButtonModule, MatButton } from '@angular/material/button';
import { buttonArgTypes } from './button-argtypes';

const meta: Meta<MatButton & { label: string; buttonType: string }> = {
  title: 'Components/Button',
  component: MatButton,
  tags: ['autodocs'],
  argTypes: {
    ...buttonArgTypes,
  },
  decorators: [
    moduleMetadata({
      imports: [MatButtonModule],
    }),
  ],
  render: (args) => ({
    props: args,
    template: `
    <button  *ngIf="buttonType === 'mat-button'" mat-button  ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-raised-button' " mat-raised-button ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-stroked-button' " mat-stroked-button ${argsToTemplate(
      args
    )} >{{label}}</button>
    <button *ngIf="buttonType === 'mat-flat-button' "  mat-flat-button ${argsToTemplate(
      args
    )} >{{label}}</button>
\    `,
  }),
};

export default meta;
type Story = StoryObj<MatButton & { label: string; buttonType: string }>;

export const accent: Story = {
  args: {
    color: 'accent',
    disabled: false,
    ariaDisabled: false,
    disabledInteractive: false,
    disableRipple: false,
    label: 'Button',
    buttonType: 'mat-button',
  },
};

export const Primary: Story = {
  args: {
    ...accent.args,
    color: 'primary',
  },
};

export const Warn: Story = {
  args: {
    ...accent.args,
    color: 'warn',
  },
};

export const Disabled: Story = {
  args: {
    ...accent.args,
    disabled: true,
  },
};

export const matRaisedButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-raised-button',
  },
};

export const matRaisedButtonPrimary: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    color: 'primary',
  },
};

export const matRaisedButtonWarn: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    color: 'warn',
  },
};

export const MatRaisedButtonDisabled: Story = {
  args: {
    ...matRaisedButtonAccent.args,
    disabled: true,
  },
};

export const MatStrokedButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-stroked-button',
  },
};

export const MatStrokedButtonPrimary: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    color: 'primary',
  },
};

export const MatStrokedButtonWarn: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    color: 'warn',
  },
};

export const MatStrokedButtonDisabled: Story = {
  args: {
    ...MatStrokedButtonAccent.args,
    disabled: true,
  },
};

export const MatFlatButtonAccent: Story = {
  args: {
    ...accent.args,
    buttonType: 'mat-flat-button',
  },
};

export const MatFlatButtonPrimary: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    color: 'primary',
  },
};

export const MatFlatButtonWarn: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    color: 'warn',
  },
};

export const MatFlatButtonDisabled: Story = {
  args: {
    ...MatFlatButtonAccent.args,
    disabled: true,
  },
};
