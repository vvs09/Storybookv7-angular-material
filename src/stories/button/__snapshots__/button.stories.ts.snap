// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Button Disabled smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mat-mdc-button mat-accent mat-mdc-button-disabled mat-mdc-button-base"
          ng-reflect-color="accent"
          mat-ripple-loader-disabled
          ng-reflect-disabled="true"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
          disabled="true"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button MatFlatButtonAccent smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-flat-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--unelevated mat-mdc-unelevated-button mat-accent mat-mdc-button-base"
          ng-reflect-color="accent"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button MatFlatButtonDisabled smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-flat-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--unelevated mat-mdc-unelevated-button mat-accent mat-mdc-button-disabled mat-mdc-button-base"
          ng-reflect-color="accent"
          mat-ripple-loader-disabled
          ng-reflect-disabled="true"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
          disabled="true"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button MatFlatButtonPrimary smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-flat-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--unelevated mat-mdc-unelevated-button mat-primary mat-mdc-button-base"
          ng-reflect-color="primary"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button MatFlatButtonWarn smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-flat-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--unelevated mat-mdc-unelevated-button mat-warn mat-mdc-button-base"
          ng-reflect-color="warn"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button MatRaisedButtonDisabled smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-raised-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--raised mat-mdc-raised-button mat-accent mat-mdc-button-disabled mat-mdc-button-base"
          ng-reflect-color="accent"
          mat-ripple-loader-disabled
          ng-reflect-disabled="true"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
          disabled="true"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button MatStrokedButtonAccent smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-stroked-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--outlined mat-mdc-outlined-button mat-accent mat-mdc-button-base"
          ng-reflect-color="accent"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button MatStrokedButtonDisabled smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-stroked-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--outlined mat-mdc-outlined-button mat-accent mat-mdc-button-disabled mat-mdc-button-base"
          ng-reflect-color="accent"
          mat-ripple-loader-disabled
          ng-reflect-disabled="true"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
          disabled="true"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button MatStrokedButtonPrimary smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-stroked-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--outlined mat-mdc-outlined-button mat-primary mat-mdc-button-base"
          ng-reflect-color="primary"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button MatStrokedButtonWarn smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-stroked-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--outlined mat-mdc-outlined-button mat-warn mat-mdc-button-base"
          ng-reflect-color="warn"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button Primary smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mat-mdc-button mat-primary mat-mdc-button-base"
          ng-reflect-color="primary"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button Warn smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mat-mdc-button mat-warn mat-mdc-button-base"
          ng-reflect-color="warn"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button accent smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mat-mdc-button mat-accent mat-mdc-button-base"
          ng-reflect-color="accent"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button matRaisedButtonAccent smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-raised-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--raised mat-mdc-raised-button mat-accent mat-mdc-button-base"
          ng-reflect-color="accent"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button matRaisedButtonPrimary smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-raised-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--raised mat-mdc-raised-button mat-primary mat-mdc-button-base"
          ng-reflect-color="primary"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;

exports[`Components/Button matRaisedButtonWarn smoke-test 1`] = `
<storybook-root ng-version="17.2.2">
  <button mat-raised-button
          mat-ripple-loader-uninitialized
          mat-ripple-loader-class-name="mat-mdc-button-ripple"
          class="mdc-button mdc-button--raised mat-mdc-raised-button mat-warn mat-mdc-button-base"
          ng-reflect-color="warn"
          ng-reflect-disabled="false"
          ng-reflect-disabled-interactive="false"
          ng-reflect-disable-ripple="false"
  >
    <span class="mat-mdc-button-persistent-ripple mdc-button__ripple">
    </span>
    <span class="mdc-button__label">
      Button
    </span>
    <span class="mat-mdc-focus-indicator">
    </span>
    <span class="mat-mdc-button-touch-target">
    </span>
  </button>
</storybook-root>
`;
